### PR TITLE
Add the inputmode attribute

### LINKED ISSUE #541: Safari on iOS brings up numeric keyboard for name field
### General information

* SDK version: 3.65.0
* Environment: Production
* Browser and OS: Safari on iOS

I'm not sure what is causing this problem, but I believe adding the [inputmode](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode) attribute should help resolve the ambiguity that Safari on iOS might be facing.

### PR DESCRIPTION
**Pull request by davidbarratt**
### Summary
Adds the [inputmode](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode) attribute to the form fields. Fixes #541.

### Checklist

- [ ] Added a changelog entry

### CODE REVIEW

**Comment by crookedneighbor** [comment]
TIL `inputmode`

I'll give this a try, but I'm not sure this makes any functional difference to the code.

File: src/hosted-fields/internal/index.js [hunk 1]
```diff
@@ -92,6 +92,7 @@ function makeMockInput(name) {
   input.type = 'text';
   input.name = name;
   input.setAttribute('autocomplete', constants.autocompleteMappings[name]);
+  input.setAttribute('inputmode', constants.inputmodeMappings[name]);
```
Comment (crookedneighbor) [inline src/hosted-fields/internal/index.js:95]: This input is hidden to accommodate autofill and should never be accessed from the user, so no need to add the attribute here.

**Comment by davidbarratt** [comment]
> I'll give this a try, but I'm not sure this makes any functional difference to the code.

It should mean you can stop using `type="tel"` which is a lie. :)

**Comment by crookedneighbor** [comment]
Sorry for the delay on this, been busy with other projects. Will try to get this in within a week or two.

**Comment by crookedneighbor** [comment]
👋  this looks reasonable to me. Gonna do some final testing after the holidays and get this merged in.

**Comment by crookedneighbor** [comment]
OK, finally got some time to test this and there's weird behavior with iOS 12. `inputMode` is _technically_ implemented in iOS 12.2, but it launches the alphanumeric keyboard for the `numeric` value. It's correctly showing the numbers, but not using the numeric keyboard. This looks like it's fixed in ios 13 and 14, so we'll definitely add this for our next major version bump when we can make breaking changes like this.

I'll leave this open with the `v4` tag.

**Comment by nidhinarendra** [comment]
The work is complete and added to the codebase

**Pull request closed without merging**